### ISSUE TITLE
Fingerprint.com

### ISSUE THREAD

**Issue by Stryder24** [arkenfox/user.js#1647]
Has anyone found a way to prevent fingerprinting on the the Fingerprint Pro Demo?   --> https://fingerprint.com/demo/

Everything that I tried did not work with Brave or Firefox.  I am using VPN and changed location several times.  I am using uBlock and arkenfox.  Since they discuss how to defeat all common fingerprint prevention attempts, and guarantee 99.5% accuracy, its not a surprise that none of this works.

When I added this **||c.fingerprint.com/v3/NIrKSr1SW3HEAoyttBe2/loader_v3.8.2.js$script,1p**  to uBlock or I disabled JavaScript, then the fingerprinting did not work, however I am pretty sure that my simple filter will not work when it is deployed in production, and disabling JavaScript altogether is not ideal.

Great news-->"Fingerprint Triples Revenue and Users, Crosses One Billion Monthly API Calls"

More details:
https://fingerprint.com/blog/browser-anti-fingerprinting-techniques/
https://fingerprint.com/blog/what-is-tls-fingerprinting-transport-layer-security/
https://fingerprint.com/blog/tag/fingerprinting/

**Comment by MightEMatt** [comment]
I was able to roll one new ID with a new firefox profile and location. I was able to roll another using Edge and a new location. I got the same roll as arkenfox on librewolf 111.0-3 on a new location. I was able to roll a new ID on librewolf 109.0.1-2 and a new location.

So as far as I can tell, arkenfox + VPN + user agent randomization would defeat it. Or use TOR and presumably get the same ID as everybody else.

**Comment by Thorin-Oakenpants** [comment]
read the [wiki](https://github.com/arkenfox/user.js/wiki/3.3-Overrides-[To-RFP-or-Not])

you are wasting your time

**Comment by Thorin-Oakenpants** [comment]
> When I added this

Don't. That's not how it works. You don't block a PoC's js file and expect it to have real world usage - and you just fucked up your own test and can no longer use it - that's silly. The fact is that the js file(s) can be named anything, and even embedded inside other site critical files. Enumerating badness is not a proper solution and easily bypassed

**Comment by Thorin-Oakenpants** [comment]
> + user agent randomization

you can't hide your engine or version or OS, os architecture, or even major OS version

> Or use TOR and presumably get the same ID as everybody else.

there is no such thing as the same FP for all tor browser users - this isn't how it works - there are still buckets of users with things you must state e.g. different languages (you can't use en-US if you can only read Arabic, for example), and things you cannot hide, e.g. different OSes, different fonts on windows (e.g. 7 vs 8 vs 10 vs 11), different screen sizes (tied to inner window) due to hardware limitations, different devicePixelRatios, subpixels in measurements, etc

**Comment by Thorin-Oakenpants** [comment]
read this thread and listen to what Peter says - `https://github.com/brave/brave-browser/issues/14031`

**Comment by KOLANICH** [comment]
>https://fingerprint.com/blog/browser-anti-fingerprinting-techniques/
>TLDR:
>* antifingerprinting is detectable
>* sometimes it is incomplete and bypassable
>* antifingerprinting causes UX issues, so only very little people use it, allowing it be used for fingerprinting and not harming the fingerprinting of majority of users

**Comment by KOLANICH** [comment]
>https://fingerprint.com/blog/what-is-tls-fingerprinting-transport-layer-security/
>TLDR:
>It is possible to identify a TLS library and to check if it matches the library used by useragents.

**Comment by Stryder24** [comment]
That engendered quite a response.   I absolutely realized that manually blocking a PoC is worthless, I wasn't presenting it as anything useful.    I thought it was interesting and I was sharing some information and my limited time playing with it (perhaps 30 minutes).  I fully understand that my knowledge of this is dwarfed by many or most in this community.  That said, I will not post any messages going forward since they do not seem to be well received.

**Comment by MightEMatt** [comment]
> you can't hide your engine or version or OS, os architecture, or even major OS version

But you _can_ randomize your UA. The goal isn't realistically to avoid immediate fingerprinting, but to appear as multiple unique visitors on subsequent attempts. Appearing as a brand new W10 user on every visit is a pretty good start. To be fair you're already one of 5 Firefox users in 2023 with RFP on requesting english websites in Zimbabwe, trying to remain anonymous is harder the harder you try.

> there is no such thing as the same FP for all tor browser users...

No need to be like that. As you said the goal is to achieve a _common_ ID that isn't unique to any one user. Being an english speaking American on TOR and _absolutely not resizing the browser_ should in theory already achieve that.

At some point you'll reach the point that you'll be using a unique tails instance on a random VPN for every website and burning the drive when you're done, and at that point SWAT is already at the door. Just use arken/libre/tor and practice good opsec and they'll get about as little information from you as they reasonably can.

**Comment by Thorin-Oakenpants** [comment]
> [tor browser] should in theory already achieve that.

nobody knows - until we do a real world study/survey with one fingerprint per profile. but sure can tell with a lot of math and science before then. I know you're not saying this, but why only care about en-US - we need to also protect the long thin tail - and metrics even with a tiny bit of entropy can have devastating effects on those smaller buckets

there are plans/ideas afoot to do this - a fingerprint test via tor browser - maybe annually - no PII collected, no cookie per user to try and limit tainted data sets, one hit per profile, entirely opt in - that is precious analytical data, imagine if 50k or 100k users replied

> But you can randomize your UA

the UA is already manifest in other metrics and isn't super stable - it is a loose metric that is almost meaningless

> and practice good opsec

you can't code meatware
